#!/usr/bin/env python
PACKAGE = "proc_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("P", double_t, 0, "X: The proportional coefficient", 0, -100.0, 100.0)
gen.add("I", double_t, 0, "X: The integration coefficient", 0, -100.0, 100.0)
gen.add("D", double_t, 0, "X: The derivative coefficient", 0, -100.0, 100.0)
gen.add("I_LIMIT", double_t, 0, "X: Maximum integration", 0, -100.0, 100.0)
gen.add("MIN_ACTUATION", double_t, 0, "X: Minimum actuation", 0, -100.0, 100.0)
gen.add("MAX_ACTUATION", double_t, 0, "X: Maximum actuation", 0, -100.0, 100.0)
gen.add("BBox", double_t, 0, "Bounding Box X for target reached, in meter", 0.3, 0, 7)
gen.add("CONSTANT_DEPTH_FORCE", double_t, 0, "Bounding Box X for target reached, in meter", 0, -20, 20)
gen.add("LPF_BETA", double_t, 1, "Bounding Box X for target reached, in meter", 0, 0, 1)


exit(gen.generate(PACKAGE, "proc_control", "Controller"))